sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java

```diff
@@ -140,23 +165,48 @@ public void subscribe(CoreSubscriber<? super T> coreSubscriber) {
                     HttpConstants.StatusCodes.OK);
             }
         }).doOnError(throwable -> {
-            if (pagedFluxOptions.getTracerProvider().isEnabled()) {
-                pagedFluxOptions.getTracerProvider().endSpan(parentContext.get(), Signal.error(throwable),
-                    TracerProvider.ERROR_CODE);
-            }
-
             if (pagedFluxOptions.getCosmosAsyncClient() != null &&
                 Configs.isClientTelemetryEnabled(BridgeInternal.isClientTelemetryEnabled(pagedFluxOptions.getCosmosAsyncClient())) &&
                 throwable instanceof CosmosException) {
                 CosmosException cosmosException = (CosmosException) throwable;
+                if (pagedFluxOptions.getTracerProvider().isEnabled()) {
+                   ((Span) parentContext.get().getData(PARENT_SPAN_KEY).get()).makeCurrent();
+                    try {
+                        addDiagnosticsOnTracerEvent(pagedFluxOptions.getTracerProvider(),
+                            cosmosException.getDiagnostics(), parentContext.get());
+                    } catch (JsonProcessingException ex) {
+                        LOGGER.debug("Error while serializing diagnostics for tracer", ex);
```

Comment:
shouldn't this be warning, here and else where?

do we expect these failure to happen frequently?

Reply:
I was debating myself, i.e. if for any reason , cx start getting this on every request it will fill the logs. However we are not expecting this at all , so lets have warning . Done